pandas/core/array_algos/take.py

```diff
@@ -121,6 +121,43 @@ def _take_nd_ndarray(
     return out
 
 
+def take_1d(
+    arr: ArrayLike,
+    indexer: np.ndarray,
+    fill_value=None,
+    allow_fill: bool = True,
+):
+    """
+    Specialized version for 1D arrays. Differences compared to take_nd:
+
+    - Assumes input (arr, indexer) has already been converted to numpy array / EA
+    - Only works for 1D arrays
+
+    To ensure the lowest possible overhead.
+
+    TODO(ArrayManager): mainly useful for ArrayManager, otherwise can potentially
+    be removed again if we don't end up with ArrayManager.
```

Comment:
conversely, if we end up with only-AM, could we remove non-this functions?

Reply:
Possibly, yes (I would need to check where take_nd is used to be sure that all such cases would become 1D)